### DISCUSSION TITLE
Multiple guards with blade authorization.

### DISCUSSION THREAD

**Discussion by j-dexx** [laravel/framework#38299] [General]
Hi,

I've got an app that uses multiple guards.  On routes without any auth middleware on blade's `@can` will use the default guard.  This means that the policies don't always work the way I'd like - if a user is logged in via a different guard.

Currently what I've decided to do is this in my auth service provider:
```php
$this->app->extend(AuthManager::class, function ($authManager, $app) {
    $authManager->resolveUsersUsing(function () {
        foreach (['guard-a', 'guard-b', 'guard-c'] as $guard) {
            if ($user = Auth::guard($guard)->user()) {
                return $user;
            }
        }
        return null;
    });
    return $authManager;
});
```

Is there a better way to accomplish this?  Or a way to set the user resolver only for blade?

**Comment by murilolivorato** [comment]
wold be a good idea to use police -
https://medium.com/@murilolivorato/getting-started-with-laravel-policies-how-to-control-access-a-step-by-step-guide-3bb080fe3762
